draft-ounsworth-csr-attestation.md

```diff
@@ -363,6 +374,62 @@ Columns:
 
 # Security Considerations
 
+A PKCS#10 or CRMF Certification Request message consists of a
+distinguished name, a public key, and optionally a set of attributes,
+collectively signed by the entity requesting certification.
+The private key used to sign the CSR MUST
+be different from the key used to sign evidence by the attester. To
+demonstrate that the private key used to sign the CSR is generated,
+stored, and used in a secure environment that has controls to prevent
+theft or misuse (including being non-exportable / non-recoverable),
+the attesting environment has to collect claims about this secure
+environment (or target environment, as shown in {{fig-attester}}).
+
+{{fig-attester}} shows the interaction inside an attester. The
+attesting environment, which is provisioned with an attestation key,
+retrieves claims about the target environment. The target
+environment offers key generation, storage and usage, which it
+makes available to services. The attesting environment collects
+these claims about the target environment and signs them and
+exports evidence for use in remote attestation via a CSR.
+
+~~~
+                   ^
+                   |CSR with
+                   |Evidence
+     .-------------+-------------.
+     |                           |
+     |       CSR Library         |<-----+
+     |                           |      |
+     '---------------------------'      |
+            |  ^         ^              |
+ Private    |  | Public  | Signature    |
+ Key        |  | Key     | Operation    |
+ Generation |  | Export  |              |
+            |  |         |              |
+ .----------|--|---------|------------. |
+ |          |  |         |    Attester| |
```

Comment:
The text above implies that the key-material for signing the CSR is in a target environment. Following that thought it would also imply that the signing code exists next to the key material in the same target environment.

Reply:
It is, of course, an implementation decision of where to put what code. In the drawing I assume that the CSR library is outside the attester but the attester performs a software measurement of that code.

I will add a note to the draft.